infrastructure/tooling/src/generate/generators/go-version.js

```diff
@@ -67,15 +70,24 @@ exports.tasks = [{
         `MIN_GO_MINOR_VERSION=${goVersionMinor}`));
 
     utils.status({ message: 'workers/generic-worker/gw-decision-task/tasks.yml' });
+    const previousGoVersion = (await readRepoFile('workers/generic-worker/gw-decision-task/tasks.yml')).match(/go[0-9.]+/)[0].substring(2).split('.');
+    const prevGoVersionMajor = previousGoVersion[0];
+    const prevGoVersionMinor = previousGoVersion[1];
+    const prevGoVersionBugfix = previousGoVersion[2] || '';
+    console.error(prevGoVersionMajor, prevGoVersionMinor, prevGoVersionBugfix);
     await modifyRepoFile('workers/generic-worker/gw-decision-task/tasks.yml',
       contents => contents.replace(
-        /go [0-9]+\.[0-9]+\.[0-9]+/g,
-        `go ${goVersionMajor}.${goVersionMinor}.${goVersionBugfix}`,
+        /go [0-9.]+/g,
+        goVersionBugfix ?
+          `go ${goVersionMajor}.${goVersionMinor}.${goVersionBugfix}` :
+          `go ${goVersionMajor}.${goVersionMinor}`,
       ).replace(
-        /gopath[0-9]+\.[0-9]+\.[0-9]+/g,
-        `gopath${goVersionMajor}.${goVersionMinor}.${goVersionBugfix}`,
+        /gopath[0-9.]+/g,
+        goVersionBugfix ?
```

Comment:
also this ternary block appears at least 3 times, maybe using an extra variable will reduce this complexity a bit

or if you have those in array you can the trick like
```js
const newVersion = semVer.filter(a => a).join('.')
```

:) sorry, just nit-picking